doc/guides/modules.rst

```diff
@@ -75,6 +75,25 @@ module:
      cmake: .
      kconfig: Kconfig
 
+
+Build system integration
+========================
+
+When a module has been described using the :file:`module.yml` file then it will
+automatically be included into the Zephyr Kconfig and CMake build system.
+
+To facilitate referencing between Zephyr and Zephyr modules, then all modules
+will be accessible through Kconfig and CMake variables.
+
+In Kconfig and CMake, the ``ZEPHYR_<module-name>_MODULE_DIR`` can be used to
```

Comment:
I didn't raise a bug about this because it wasn't officially documented before, but this is not really correct.

The existence of a module.yml file does not guarantee this variable is defined, even in CMake.

See this source code comment in one of my PRs for details: https://github.com/zephyrproject-rtos/zephyr/pull/27688/files#diff-a1f9e35746e0274b9ad8cfb12162414dR38

I think this situation should be fixed, or at least documented more clearly.

Reply:
Agreed, as minimum this should be rephrased to be more precise that only if build files are present in the module.

Guess the introduction of sanity in `module.yml` made a small hole here.

Reply:
> as minimum this should be rephrased to be more precise that only if build files are present in the module.

I agree that would be the minimum acceptable change, since it makes the documentation correct.

Reply:
(Although FTR I would appreciate it if the existence of a module.yml resulted in the definition of ZEPHYR_FOO_MODULE_DIR. That is the best possible world but I agree it's outside the scope of this PR)

Reply:
couldn't leave code like that.
Fixed. Now docs and code is aligned.

Reply:
OK, rebased and tested on top of the updated version. I confirm the fix. Thanks for fixing :+1: